docs/tutorial/quickstart.md

```diff
@@ -0,0 +1,282 @@
+# Tutorial: Get started with oras-go v2
+
+This tutorial introduces the basics of managing OCI artifacts with the [oras-go v2](https://pkg.go.dev/oras.land/oras-go/v2) package.
+
+You'll get the most out of this tutorial if you have a basic familiarity with Go and its tooling. If this is your first exposure to Go, please see [Tutorial: Get started with Go](https://golang.org/doc/tutorial/getting-started) for a quick introduction.
+
+The tutorial includes the following sections:
+
+1. Connect to a remote repository.
+2. Show tags in the repository.
+3. Push layers to the repository.
+4. Push a manifest to the repository.
+5. Fetch the manifest from the repository.
+6. Pull the manifest from the repository.
+
+The complete code is provided at the end of this tutorial.
+
+## Connect to a remote repository with basic authentication
+
+The following code snippet demonstrates how to use [NewRepository](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#NewRepository) from the [remote](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote) package to connect to a remote repository. Basic authentication (using a username and password) is handled by the [auth](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote/auth) package. Other authentication methods, such as access token authentication, are also supported.
+
+```
+ctx := context.Background()
+registry := "myregistry.example.io"
+repository := "myrepo"
+repo, err := remote.NewRepository(fmt.Sprintf("%s/%s", registry, repository))
+if err != nil {
+	panic(err)
+}
+// Note: The below code can be omitted if authentication is not required.
+repo.Client = &auth.Client{
+	Client: retry.DefaultClient,
+	Cache:  auth.NewCache(),
+	Credential: auth.StaticCredential(registry, auth.Credential{
+		Username: "username",
+		Password: "password",
+	}),
+}
+```
+
+## Show tags in the repository
+The following code snippet uses the [(*Repository) Tags](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.Tags) method from the [remote](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote) package to list the tags in the repository.
+
+```
+err = repo.Tags(ctx, "", func(tags []string) error {
+	for _, tag := range tags {
+		fmt.Println(tag)
+	}
+	return nil
+})
+if err != nil {
+	panic(err)
+}
+```
+
+## Push layers to the repository
+
+Before pushing a manifest, the layers it references must already exist in the repository. The following code snippet demonstrates how to push a config layer and a manifest layer using the [(*Repository) Push](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.Push) method.
+
+```
+// push config layer
+configLayer := []byte("example config layer")
+configLayerDescriptor := content.NewDescriptorFromBytes(ocispec.MediaTypeImageLayer, configLayer)
```

Comment:
nit: How about using `ocispec.MediaTypeImageConfig` as the config media type?

Reply:
changed

Reply:
Removed this part after using `Pack`.